Replace jest with vitest

duzda opened the pull request:
Removes also all auxiliary files.
Most changes consist of changing imports.
Also disable .js files in the project

carma12 reviewed (changes requested):
Overall LGTM. Just some questions/notes before providing an ack:
- This is just for the record (in case other contributors and developers check this PR after it is merged): it is necessary to update the `node` version using nvm to not getting any errors due to an old version, via `nvm use <version>`. In my case, I executed `nvm use 23`
- Probably the `@testing-library/jest-dom` library needs to be removed as well? If so, the `globals.d.ts` and `setupTests.ts` files should be removed or altered as well
- The `README` file should be adjusted to the new changes, as it still refers to Jest when mentioning unit tests
- In the `tsconfig.json` and `BreadCrumb.test.tsx` file there are comments mentioning Jest, so those should be removed as well

duzda commented:
Thanks, regarding the changes:
- Added `.nvmrc` file, with `lts/jod` version, which is an alias for node 22, the same as latest Fedora 41 and RHEL10 have. Also changed docs accordingly, we no longer wanna have developers running `npm` from Vagrant.
- `@testing-library/jest-dom`, strangely no, this is correct. I have just changed the imports for `vitest` instead of `jest`. jest-dom is fine with vitest, it's just a library providing a few functions we use, even though the name hints for it's removal, `setupTests.ts` has been updated however `globals.d.ts` can be safely removed.
- I've cleaned `tsconfig.json` and `BreadCrumb.test.tsx`

Please doublecheck whether everything is fine.

carma12 commented on README.md:
````diff
@@ -114,14 +114,14 @@ The integration tests use the **.feature** suffix and can be found in the _tests
 
 ```bash
-$ npm run test
+$ npm test
````
This is still `npm run test` :)

duzda replied:
`npm test` works as well? It's one of those shortcuts, such as `npm start`

carma12 replied:
Oh, you are right! I though this will require explicitly the `run` command in order to work, but I was wrong. Please ignore this message 😅

carma12 reviewed (approved):
Apart from what's mentioned in the comments, it seems to work fine after installing the vitest dependencies. LGTM.

duzda commented on README.md:
````diff
@@ -49,15 +49,15 @@ vagrant ssh-config | grep HostName | awk '{print $2}'
+
+```bash
+$ nvm install
+$ nvm use
````
No need to specify, it looks up to `.nvmrc` file, which contains the LTS version used in RHEL10 and Fedora41.

carma12 replied:
I see. Thank you for clarify this 👍🏻
